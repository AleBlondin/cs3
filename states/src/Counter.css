.counter {
  border: 1px solid black;
  width: 50px;
  height: 100px;

  justify-content: space-around;
  display: flex;
  flex-direction: column;
}

.counter + .counter {
  margin-left: 10px;
}
