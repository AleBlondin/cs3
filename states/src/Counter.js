import { useDispatch, useSelector } from "react-redux";
import "./Counter.css";
import { incrementCounter, selectCount } from "./redux";

const Counter = ({ counterId }) => {
  const dispatch = useDispatch();
  const count = useSelector(selectCount)({ counterId });

  return (
    <div className="counter">
      <button
        onClick={() => {
          dispatch(incrementCounter({ counterId }));
        }}
      >
        Click me
      </button>
      <div>{count}</div>
    </div>
  );
};

export default Counter;
