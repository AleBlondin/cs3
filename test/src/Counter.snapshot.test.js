import React from "react";
import renderer from "react-test-renderer";
import Counter from "./Counter";

it("renders correctly", () => {
  const tree = renderer.create(<Counter />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with a title", () => {
  const tree = renderer.create(<Counter title="custom title" />).toJSON();
  expect(tree).toMatchSnapshot();
});
