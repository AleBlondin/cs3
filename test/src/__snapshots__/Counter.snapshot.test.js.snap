// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="App"
  >
    Some content
  </div>,
  <button
    onClick={[Function]}
  >
    Click me
  </button>,
  "0",
]
`;

exports[`renders correctly with a title 1`] = `
Array [
  <div
    className="App"
  >
    Some content
  </div>,
  <button
    onClick={[Function]}
  >
    Click me
  </button>,
  "0",
]
`;
